src/utils/Uri.cpp

```diff
@@ -25,9 +25,19 @@ Uri::Uri(const std::string &in): path(), query(), identifier() {
 	path = p;
 }
 
-// TODO directory traversal detection
 void Uri::appendPath(const std::string &in) {
 	std::string p = in;
+
+	// check if path is '.' or '..' and erase it
+	std::string::size_type dot = p.find('.');
+	std::string::size_type dots = p.find("..");
+	if (dot == 0) {
+		p.erase(dot, 1);
+	}
+	if (dots == 0) {
+		p.erase(dots, 2);
+	}
+
```

Comment:
make seperate (static function or bool in parameter) check that only applies in the parser. directory traversal is used in config options.

Reply:
(also check with telnet and nginx if it should be 400'ed or erased)